src/ixdat/backends/directory_backend.py

```diff
@@ -0,0 +1,158 @@
+import json
+import numpy as np
+from .memory_backend import BackendBase
+from ..config import STANDARD_DATA_DIRECTORY, STANDARD_SUFFIX, STANDARD_DATA_SUFFIX
+
+
+def id_from_path(path):
+    """Return the id (int) of the row represented by given path to an ixdat file"""
+    try:
+        return int(path.stem.split("_")[0])
+    except ValueError:
+        return None
+
+
+def name_from_path(path):
+    """Return the name (str) of the row represented by given path to an ixdat file"""
+    return "".join(path.stem.split("_")[1:])
+
+
+class DirBackend(BackendBase):
+    """A database backend that loads and saves .ix files from a directory"""
+
+    def __init__(
+        self,
+        directory=STANDARD_DATA_DIRECTORY,
+        suffix=STANDARD_SUFFIX,
+        data_suffix=STANDARD_DATA_SUFFIX,
+    ):
+        """Initialize a directory database backend with the directory as Path
+
+        Args:
+            directory (Path): the directory to save files (under table subfolders)
+            suffix (str): The suffix to use for JSON-formatted metadata files
+            data_suffix (str): The suffix to use for numpy-formatted data files
+        """
+        self.directory = directory
+        self.suffix = suffix
+        self.data_suffix = data_suffix
+
+    @property
+    def name(self):
+        return f"DirBackend({self.directory})"
+
+    def save(self, obj):
+        """Save the Saveable object as a file corresponding to a row in a table"""
+        if obj.data_objects:
+            # save any data objects first as this may change the references
+            for data_obj in obj.data_objects:
+                self.save_data_obj(data_obj)
+        try:
+            table_name = obj.table_name
+        except AttributeError:
+            table_name = str(type(obj))
+        obj_as_dict = obj.as_dict()
+        obj.id = self.add_row(obj_as_dict, table_name=table_name)
+        obj.backend_name = self.name
+        return obj.id
+
+    def open(self, cls, i):
```

Comment:
why will that break linters? Will save that eventual refactoring for later. Perhaps a good time though to get a commit hook linting up and running.

Reply:
Well. It will not break them, but just make them complain because they almost all have the rule that variable length should be >=2, unless explicitly silenced for good reason e.g. x and y, it might actually be that that i is allowed as well as a usual index shorthand, but I don't think so, in any case it doesn't change the following.Of course id is a function in the global namespace so it will also complain about that. The usual way to get around that is to attach a _, so `id_`. I prefer this for readability purely because it actually still says "id".

Reply:
Oh and yeah commit hooks are awesome just don't make them to restrictive to begin with, maybe just a linter is fine.

Reply:
I can see you're right but it's annoying ;) . Will change to _id later, for efficiency at the same time as changing it to a GUID... I do want to talk about GUID implementation once more before that. And perhaps give it a spin without GUID first.

Reply:
TODO added to the class docstring.

Reply:
I like your plan for the guid stuff, that is try out the implementation without it and implement that on its own, so we can see that change in implementation on its own.

WRT to naming variables, I know this is a nit-pick and for that I apologize, but personally I prefer `id_` over `_id` because the second can be confused with something "private" (also as per pep8 https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles). But of course if you go ahead with guid you might name it guid and avoid the issue altogether ;)